curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md

```diff
@@ -1,28 +1,33 @@
 ---
 id: 657e2ecee9f60092c89338d9
-title: Step 84
+title: Step 85
 challengeType: 0
-dashedName: step-84
+dashedName: step-85
 ---
 
 # --description--
 
-If the user rolls a small straight, then the `Small straight` radio button option should be enabled with a score of `30` points.
+If the user rolls a small straight, then the `Small straight` radio button option should be enabled with a score of `30` points. A small straight consists of four separate dice rolls arranged in numerical order. 
 
-Start by creating an `if` statement to check if `uniqueNumbersStr` is included in the `smallStraightsArr`. If so, call the `updateRadioOption` with `3` and `30` for the arguments. 
+To get started, iterate through the `smallStraightsArr`. You are free to use whichever loop you prefer. If you are using a higher order function like a `forEach` or the `some` method, the callback function should take an argument of `straight`. If you are using a `for...of` loop, then your loop variable should be called `straight`.
+
+Inside the loop,  if `uniqueNumbersStr` has a combination that matches a small straight, call the `updateRadioOption` with `3` and `30` for the arguments. 
 
 # --hints--
 
-You should use the `includes` method to check if `uniqueNumbersStr` is included in the `smallStraightsArr`.
+You should use the `includes` method to check if `straight` is included in the `uniqueNumbersStr`. 
 
 ```js
-assert.match(code, /if\s*\(\s*smallStraightsArr\.includes\(\s*uniqueNumbersStr\s*\)\s*\)\s*{?/);
+assert.match(code, /\s*i?f?\(?uniqueNumbersStr\.includes\((straight|smallStraightsArr\[i\])\)\)\)?\s*\{/)  
```

Comment:
the test doesn't pass when there's some spaces before and after the arguments

```suggestion
assert.match(code, /\s*i?f?\(\s*uniqueNumbersStr\.includes\(\s*(straight|smallStraightsArr\[i\])\s*\)\s*\)\s*{/)
```

Reply:
Hi there. Would this version be acceptable ? 



```assert.match(code, /\s*i?f?\(?\s*uniqueNumbersStr\.includes\(\s*(straight|smallStraightsArr\[i\])\s*\)\)\s*\)?\s*\{/)``` 

The version of regex you gave does not work with the solution I set on the step 86 and 87.

Reply:
@a2937, the code i wrote was for only step 85. In the above context, as long as gaps before/after arguments are accepted, then any regex would work well. come up with a solution I'll test it and review it again for sure. also drop your twitter if you have one?

Reply:
I do not have a Twitter and don't plan on getting one until it is an requirement. This is part of the seeded code I added to steps 86 and 87. 

` if (smallStraightsArr.some(straight => uniqueNumbersStr.includes(straight))) {
    updateRadioOption(3, 30);
  }`

I've tested this block and it seems to work pretty well.

Reply:
commit it on the files, I'll check it from vs code

Reply:
It's been committed.